upnp/src/genlib/net/http/httpreadwrite.c

```diff
@@ -465,7 +465,7 @@ int http_SendMessage(SOCKINFO *info, int *TimeOut, const char *fmt, ...)
 		if (c == 'I') {
 			Instr = va_arg(argp, struct SendInstruction *);
 			if (Instr->ReadSendSize >= 0)
-				amount_to_be_read = (size_t)Instr->ReadSendSize;
+				amount_to_be_read = Instr->ReadSendSize;
```

Comment:
This is where the problem happend before, right? But I don't understand why a file with size 2GB + 5 bytes doesn't read as 5 bytes only then. (Or did I misunderstand the bug report?)

Reply:
Yes, I think that it would be truncated to 5 bytes. It's truncated to whatever is in the low 32 bits, this is what I am seeing before the modification.

Reply:
ok, so the bug report seems to be unprecise here.